File: src/main/java/fastcampus/innercircle/onboarding/survey/repository/SurveyQuestionRepository.java
package fastcampus.innercircle.onboarding.survey.repository;

import fastcampus.innercircle.onboarding.survey.domain.SurveyQuestion;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface SurveyQuestionRepository extends JpaRepository<SurveyQuestion, Long> {
}

File: src/main/kotlin/innerCircle/survey/repository/SurveyQuestionRepository.kt
package innerCircle.survey.repository

import innerCircle.survey.entity.SurveyQuestion
import innerCircle.survey.entity.SurveyVersion
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.time.LocalDateTime

@Repository
interface SurveyQuestionRepository : JpaRepository<SurveyQuestion, Long>{

    fun findBySurveyVersion(surveyVersion: SurveyVersion): List<SurveyQuestion>

    @Modifying
    @Query("UPDATE SurveyQuestion q SET q.deletedAt = :deletedAt WHERE q.id IN :ids")
    fun markAsDeleted(@Param("ids") ids: List<Long>, @Param("deletedAt") deletedAt: LocalDateTime)
}

